pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java

```diff
@@ -152,6 +152,20 @@ protected void delete(String path) throws MetadataStoreException {
         return cache.delete(path);
     }
 
+    protected CompletableFuture<Void> deleteIfExistsAsync(String path) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        cache.delete(path).whenComplete((ignore, ex) -> {
```

Comment:
👍 We could also do a check for `exists()` before the delete.

Reply:
Updated